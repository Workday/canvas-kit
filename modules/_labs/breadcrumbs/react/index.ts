import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
