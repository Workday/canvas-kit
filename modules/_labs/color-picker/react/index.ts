export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
