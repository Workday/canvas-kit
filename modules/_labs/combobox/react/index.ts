import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
