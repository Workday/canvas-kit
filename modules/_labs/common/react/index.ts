export * from './lib/Flex';
export * from './lib/Box';
