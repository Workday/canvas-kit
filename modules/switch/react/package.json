{
  "name": "@workday/canvas-kit-react-switch",
  "version": "3.6.0",
  "description": "A Canvas-styled switch",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/switch/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "watch": "yarn build:es6 -w",
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "switch"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@workday/canvas-kit-labs-react-core": "^3.6.0",
    "@workday/canvas-kit-react-common": "^3.6.0",
    "@workday/canvas-kit-react-core": "^3.6.0",
    "uuid": "^3.3.3"
  }
}
