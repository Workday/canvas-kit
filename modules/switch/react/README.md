# Canvas Kit Switch

A pill shaped switch. This is an
[_controlled_](https://reactjs.org/docs/forms.html#controlled-components) `input` component.
Undocumented props are spread to the `input` element.

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-switch
```

## Usage

#### Simple Example

**Note:** While a base switch component is provided in this package, it is **not accessible** when
used as is. It should be used in tandem with [`FormField`](../../form-field/react) to be made fully
accessible (see below).

```tsx
import * as React from 'react';
import {Switch} from '@workday/canvas-kit-react-switch';

<Switch disabled={false} checked={checked} onChange={this.handleCheck} />;
```

#### Accessible Example

```tsx
import * as React from 'react';
import {Switch} from '@workday/canvas-kit-react-switch';
import FormField from '@workday/canvas-kit-react-form-field';

<FormField label="My Field" inputId="my-switch-field">
  <Switch disabled={false} checked={checked} onChange={this.handleCheck} id="my-switch-field" />;
</FormField>;
```

If use inside a FormField doesn't work for your use case, you can use the `aria-labelledBy`
attribute.

```tsx
import * as React from 'react';
import {Switch} from '@workday/canvas-kit-react-switch';
<label id="123">Label</label>
...
<Switch disabled={false} checked={checked} onChange={this.handleCheck} aria-labelledBy="123" />;
```

## Static Properties

> None

## Component Props

### Required

> None

### Optional

#### `checked: boolean`

> Whether or not the switch is on (`true`) or off (`false`)

Default: `false`

---

#### `disabled: boolean`

> Whether or not the switch is disabled (not able to be switched on or off)

Default: `false`

---

#### `id: string`

> The HTML attribute `id` for the underlying input checkbox component.

---

#### `onChange: (e: React.SyntheticEvent<HTMLInputElement>) => void`

> A callback that gets called everytime the switch checked state changes.

---

#### `value: string`

> The `value` attribute of the input checkbox.

---

#### `inputRef: React.Ref<HTMLInputElement>`

> A ref to the underlying input checkbox element. Use this to imperatively switch or focus this
> component.

---

#### `error: ErrorType`

> The type of error to display, if any.

| Type  | Description                     |
| ----- | ------------------------------- |
| Error | Red outline with error icon.    |
| Alert | Yellow outline with alert icon. |

Default: `undefined`
