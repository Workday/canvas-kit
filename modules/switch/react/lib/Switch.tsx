import * as React from 'react';
import styled from 'react-emotion';
import {focusRing} from '@workday/canvas-kit-react-common';
import {colors, depth} from '@workday/canvas-kit-react-core';

export interface SwitchProps extends React.InputHTMLAttributes<HTMLInputElement> {
  checked: boolean;
  disabled?: boolean;
  id?: string;
  onChange?: (e: React.SyntheticEvent) => void;
  value?: string;
  inputRef?: React.Ref<HTMLInputElement>;
}

const circleSize = 12;
const switchWidth = 32;
const switchHeight = 16;
const translateLength = switchWidth - switchHeight;

const SwitchContainer = styled('div')({
  position: 'relative',
  height: switchHeight,
  width: switchWidth,
});

const SwitchInput = styled('input')<SwitchProps>(
  {
    position: 'absolute',
    height: switchHeight,
    width: switchWidth,
    margin: 0,
    borderRadius: 999,
    '&:focus, &:active': {
      outline: 'none',
      ...focusRing(2, 2),
    },
  },
  ({disabled}) => ({
    cursor: disabled ? 'not-allowed' : 'pointer',
  })
);

const SwitchBackground = styled('div')<Pick<SwitchProps, 'checked' | 'disabled'>>(
  {
    boxSizing: 'border-box',
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    pointerEvents: 'none',
    width: switchWidth,
    height: switchHeight,
    borderRadius: 999,
    padding: '0px 2px',
    transition: 'background-color 200ms ease',
  },
  ({checked, disabled}) => ({
    backgroundColor: disabled ? colors.soap400 : checked ? colors.blueberry500 : colors.licorice200,
  })
);

const SwitchCircle = styled('div')<Pick<SwitchProps, 'checked'>>(({checked}) => ({
  width: circleSize,
  height: circleSize,
  borderRadius: 999,
  ...depth[1],
  backgroundColor: colors.frenchVanilla100,
  transform: checked ? `translateX(${translateLength}px)` : 'translateX(0)',
  transition: 'transform 150ms ease',
  pointerEvents: 'none',
}));

export default class Switch extends React.Component<SwitchProps> {
  public static defaultProps = {
    checked: false,
  };

  public render() {
    // TODO: Standardize on prop spread location (see #150)
    const {checked, disabled, id, inputRef, onChange, value, ...elemProps} = this.props;

    return (
      <SwitchContainer>
        <SwitchInput
          checked={checked}
          disabled={disabled}
          id={id}
          innerRef={inputRef}
          onChange={onChange}
          role="checkbox"
          tabIndex={0}
          type="checkbox"
          value={value}
          {...elemProps}
        />
        <SwitchBackground checked={checked} disabled={disabled}>
          <SwitchCircle checked={checked} />
        </SwitchBackground>
      </SwitchContainer>
    );
  }
}
