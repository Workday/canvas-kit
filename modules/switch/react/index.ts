import Switch from './lib/Switch';

export default Switch;
export {Switch};
export * from './lib/Switch';
