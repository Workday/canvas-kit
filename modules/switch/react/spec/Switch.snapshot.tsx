import * as React from 'react';
import Switch from '../lib/Switch';
import * as renderer from 'react-test-renderer';

describe('Switch Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Switch checked={false} onChange={jest.fn()} />);
    expect(component).toMatchSnapshot();
  });
  test('renders switch on', () => {
    const component = renderer.create(<Switch checked={true} onChange={jest.fn()} />);
    expect(component).toMatchSnapshot();
  });
  test('renders switch disabled unchecked', () => {
    const component = renderer.create(
      <Switch disabled={true} checked={false} onChange={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders switch disabled checked', () => {
    const component = renderer.create(
      <Switch disabled={true} checked={true} onChange={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders switch with name and value', () => {
    const component = renderer.create(
      <Switch value="user" disabled={false} checked={true} onChange={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders switch with id', () => {
    const component = renderer.create(
      <Switch value="user" id={'123'} disabled={false} checked={true} onChange={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders a switch with custom prop', () => {
    const component = renderer.create(
      <Switch
        title="hello world"
        value="user"
        id={'123'}
        disabled={false}
        checked={true}
        onChange={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
