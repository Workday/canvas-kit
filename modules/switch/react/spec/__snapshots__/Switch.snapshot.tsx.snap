// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Snapshots renders a switch with custom prop 1`] = `
.emotion-6 {
  position: relative;
  height: 32px;
  width: 32px;
}

.emotion-0 {
  position: absolute;
  height: 32px;
  width: 32px;
  margin: 0;
  margin-left: 4px;
  border-radius: 999px;
  opacity: 0;
  cursor: pointer;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type {
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:focus ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:active ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

.emotion-4 {
  box-sizing: border-box;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  margin-top: 8px;
  width: 32px;
  height: 16px;
  border-radius: 999px;
  padding: 0px 2px;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  background-color: #005cb9;
}

.emotion-2 {
  width: 12px;
  height: 12px;
  border-radius: 999px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
  -webkit-transition: -webkit-transform 150ms ease;
  -webkit-transition: transform 150ms ease;
  transition: transform 150ms ease;
  pointer-events: none;
}

<div
  className="emotion-6 emotion-7"
>
  <input
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    id="123"
    onChange={[MockFunction]}
    role="checkbox"
    tabIndex={0}
    title="hello world"
    type="checkbox"
    value="user"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
    disabled={false}
  >
    <div
      checked={true}
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;

exports[`Switch Snapshots renders as expected 1`] = `
.emotion-6 {
  position: relative;
  height: 32px;
  width: 32px;
}

.emotion-0 {
  position: absolute;
  height: 32px;
  width: 32px;
  margin: 0;
  margin-left: 4px;
  border-radius: 999px;
  opacity: 0;
  cursor: pointer;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type {
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:focus ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:active ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

.emotion-4 {
  box-sizing: border-box;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  margin-top: 8px;
  width: 32px;
  height: 16px;
  border-radius: 999px;
  padding: 0px 2px;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  background-color: #7b858f;
}

.emotion-2 {
  width: 12px;
  height: 12px;
  border-radius: 999px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 150ms ease;
  -webkit-transition: transform 150ms ease;
  transition: transform 150ms ease;
  pointer-events: none;
}

<div
  className="emotion-6 emotion-7"
>
  <input
    checked={false}
    className="emotion-0 emotion-1"
    id="123"
    onChange={[MockFunction]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={false}
    className="emotion-4 emotion-5"
  >
    <div
      checked={false}
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;

exports[`Switch Snapshots renders switch disabled checked 1`] = `
.emotion-6 {
  position: relative;
  height: 32px;
  width: 32px;
}

.emotion-2 {
  width: 12px;
  height: 12px;
  border-radius: 999px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
  -webkit-transition: -webkit-transform 150ms ease;
  -webkit-transition: transform 150ms ease;
  transition: transform 150ms ease;
  pointer-events: none;
}

.emotion-0 {
  position: absolute;
  height: 32px;
  width: 32px;
  margin: 0;
  margin-left: 4px;
  border-radius: 999px;
  opacity: 0;
  cursor: not-allowed;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type {
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:focus ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:active ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

.emotion-4 {
  box-sizing: border-box;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  margin-top: 8px;
  width: 32px;
  height: 16px;
  border-radius: 999px;
  padding: 0px 2px;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  background-color: #dfe2e6;
}

<div
  className="emotion-6 emotion-7"
>
  <input
    checked={true}
    className="emotion-0 emotion-1"
    disabled={true}
    id="123"
    onChange={[MockFunction]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
    disabled={true}
  >
    <div
      checked={true}
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;

exports[`Switch Snapshots renders switch disabled unchecked 1`] = `
.emotion-6 {
  position: relative;
  height: 32px;
  width: 32px;
}

.emotion-2 {
  width: 12px;
  height: 12px;
  border-radius: 999px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 150ms ease;
  -webkit-transition: transform 150ms ease;
  transition: transform 150ms ease;
  pointer-events: none;
}

.emotion-0 {
  position: absolute;
  height: 32px;
  width: 32px;
  margin: 0;
  margin-left: 4px;
  border-radius: 999px;
  opacity: 0;
  cursor: not-allowed;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type {
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:focus ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:active ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

.emotion-4 {
  box-sizing: border-box;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  margin-top: 8px;
  width: 32px;
  height: 16px;
  border-radius: 999px;
  padding: 0px 2px;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  background-color: #dfe2e6;
}

<div
  className="emotion-6 emotion-7"
>
  <input
    checked={false}
    className="emotion-0 emotion-1"
    disabled={true}
    id="123"
    onChange={[MockFunction]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={false}
    className="emotion-4 emotion-5"
    disabled={true}
  >
    <div
      checked={false}
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;

exports[`Switch Snapshots renders switch on 1`] = `
.emotion-6 {
  position: relative;
  height: 32px;
  width: 32px;
}

.emotion-0 {
  position: absolute;
  height: 32px;
  width: 32px;
  margin: 0;
  margin-left: 4px;
  border-radius: 999px;
  opacity: 0;
  cursor: pointer;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type {
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:focus ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:active ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

.emotion-4 {
  box-sizing: border-box;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  margin-top: 8px;
  width: 32px;
  height: 16px;
  border-radius: 999px;
  padding: 0px 2px;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  background-color: #005cb9;
}

.emotion-2 {
  width: 12px;
  height: 12px;
  border-radius: 999px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
  -webkit-transition: -webkit-transform 150ms ease;
  -webkit-transition: transform 150ms ease;
  transition: transform 150ms ease;
  pointer-events: none;
}

<div
  className="emotion-6 emotion-7"
>
  <input
    checked={true}
    className="emotion-0 emotion-1"
    id="123"
    onChange={[MockFunction]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
  >
    <div
      checked={true}
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;

exports[`Switch Snapshots renders switch with id 1`] = `
.emotion-6 {
  position: relative;
  height: 32px;
  width: 32px;
}

.emotion-0 {
  position: absolute;
  height: 32px;
  width: 32px;
  margin: 0;
  margin-left: 4px;
  border-radius: 999px;
  opacity: 0;
  cursor: pointer;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type {
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:focus ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:active ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

.emotion-4 {
  box-sizing: border-box;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  margin-top: 8px;
  width: 32px;
  height: 16px;
  border-radius: 999px;
  padding: 0px 2px;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  background-color: #005cb9;
}

.emotion-2 {
  width: 12px;
  height: 12px;
  border-radius: 999px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
  -webkit-transition: -webkit-transform 150ms ease;
  -webkit-transition: transform 150ms ease;
  transition: transform 150ms ease;
  pointer-events: none;
}

<div
  className="emotion-6 emotion-7"
>
  <input
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    id="123"
    onChange={[MockFunction]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
    value="user"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
    disabled={false}
  >
    <div
      checked={true}
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;

exports[`Switch Snapshots renders switch with name and value 1`] = `
.emotion-6 {
  position: relative;
  height: 32px;
  width: 32px;
}

.emotion-0 {
  position: absolute;
  height: 32px;
  width: 32px;
  margin: 0;
  margin-left: 4px;
  border-radius: 999px;
  opacity: 0;
  cursor: pointer;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type {
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:focus ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:focus ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

[data-whatinput="mouse"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="touch"] .emotion-0:active ~ div:first-of-type,
[data-whatinput="pointer"] .emotion-0:active ~ div:first-of-type {
  -webkit-animation: none;
  animation: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #0875e1;
}

.emotion-4 {
  box-sizing: border-box;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  margin-top: 8px;
  width: 32px;
  height: 16px;
  border-radius: 999px;
  padding: 0px 2px;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  background-color: #005cb9;
}

.emotion-2 {
  width: 12px;
  height: 12px;
  border-radius: 999px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
  -webkit-transition: -webkit-transform 150ms ease;
  -webkit-transition: transform 150ms ease;
  transition: transform 150ms ease;
  pointer-events: none;
}

<div
  className="emotion-6 emotion-7"
>
  <input
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    id="123"
    onChange={[MockFunction]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
    value="user"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
    disabled={false}
  >
    <div
      checked={true}
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;
